app/models/notify_user/aggregator.rb

```diff
@@ -0,0 +1,64 @@
+module NotifyUser
+  class Aggregator
+    def initialize(notification, aggregation_intervals)
+      @notification = notification
+      @aggregation_intervals = aggregation_intervals
+    end
+
+    def has_pending_deliveries?
+      if notification.class.aggregate_grouping?
+        unread_unsent_notifications_for_group.any?
+      else
+        unread_unsent_notifications.any?
+      end
+    end
+
+    def delay_time_in_seconds
+      delay_time_in_minutes = aggregation_intervals[aggregation_interval] || aggregation_intervals.last
```

Comment:
Could perhaps rename the `aggregration_interval` method something like `next_interval_index` - the fact that it's singular of some other thing is pretty confusing